### DISCUSSION TITLE
Why do we build in Typescript, Why not JavaScript?

### DISCUSSION THREAD

**Discussion by PareshNeosoft** [NeoSOFT-Technologies/rest-node-nestjs#142] [Ideas]
Why do we build in Typescript, Why not JavaScript?

**Comment by shantanuneosoft** [comment]
Since JavaScript is a subset of TypeScript, you can use all JavaScript libraries and code that you want in your TypeScript code.


![0ufu1q21 js-ts](https://user-images.githubusercontent.com/78724184/130910170-b3a3c111-7e48-48fb-b3f3-28064d31b78a.jpg)

**Reply by aartipawar30** [reply]
Typescript is a superset of javascript i.e it is Javascript + some additional feature(ES6). The advantage of typescript is that static typing means it always points out the compilation error at the time of development only. Because of this at the time of run time the chance of getting error are very less whereas JS is interpreted language.
TS is known as OOPS language and JS is scripting language
TS support for module whereas JS does not support module
TS has interface but JS does not have interface
Disadv. of TS is - It take time to compile the code.
